src/pages/Application/issue/EnterBTCAmount.tsx

```diff
@@ -0,0 +1,387 @@
+  const [dustValue, setDustValue] = React.useState('0');
+  const [vaultMaxAmount, setVaultMaxAmount] = React.useState('');
+
+  const [submitStatus, setSubmitStatus] = React.useState(STATUSES.IDLE);
+  const [submitError, setSubmitError] = React.useState<Error | null>(null);
+
+  React.useEffect(() => {
+    if (!polkaBTCAmount) return;
+    if (!btcToDotRate) return;
+    if (!depositRate) return;
+    if (!feeRate) return;
+    if (!vaults) return;
+
+    const bigPolkaBTCAmount = new Big(polkaBTCAmount);
+    const theBridgeFee = bigPolkaBTCAmount.mul(feeRate);
+    setBridgeFee(theBridgeFee.toString());
+
+    const theSecurityDeposit = bigPolkaBTCAmount.mul(btcToDotRate).mul(depositRate);
+    setSecurityDeposit(theSecurityDeposit.round(8).toString());
+
+    const theTotalBTCAmount = bigPolkaBTCAmount.add(theBridgeFee);
+    setTotalBTCAmount(theTotalBTCAmount.toString());
+
+    const polkaBTCAmountAsSAT = btcToSat(polkaBTCAmount);
+    const vaultId = getRandomVaultIdWithCapacity(Array.from(vaults || new Map()), new BN(polkaBTCAmountAsSAT));
+    setVaultId(vaultId ?? '');
+  }, [
+    polkaBTCAmount,
+    btcToDotRate,
+    depositRate,
+    feeRate,
+    vaults
+  ]);
+
+  React.useEffect(() => {
+    if (!polkaBtcLoaded) return;
+
+    (async () => {
+      try {
+        setStatus(STATUSES.PENDING);
+        const [
+          theFeeRate,
+          theDepositRate,
+          issuePeriodInBlocks,
+          dustValueAsSatoshi,
+          btcToDot,
+          theVaults
+        ] = await Promise.all([
+          // Loading this data is not strictly required as long as the constantly set values did
+          // not change. However, you will not see the correct value for the security deposit.
+          // But not having this data will NOT block the requestIssue
+          window.polkaBTC.issue.getFeeRate(),
+          window.polkaBTC.fee.getIssueGriefingCollateralRate(),
+          window.polkaBTC.issue.getIssuePeriod(),
+          window.polkaBTC.redeem.getDustValue(),
+          window.polkaBTC.oracle.getExchangeRate(),
+          // This data (the vaults) is strictly required to request issue
+          window.polkaBTC.vaults.getVaultsWithIssuableTokens()
+        ]);
+        setStatus(STATUSES.RESOLVED);
+
+        setFeeRate(theFeeRate);
+        setDepositRate(theDepositRate);
+        const issuePeriod = new BN(issuePeriodInBlocks.toString()).mul(new BN(BLOCK_TIME)).toNumber();
+        dispatch(updateIssuePeriodAction(issuePeriod));
+        const theDustValue = satToBTC(dustValueAsSatoshi.toString());
+        setDustValue(theDustValue);
+        setBtcToDotRate(btcToDot);
+
+        // ray test touch <
+        let theVaultMaxAmount = new BN(0);
+        // TODO: double-check
+        // The first item is the vault with the largest capacity
+        for (const issuableTokens of theVaults.values()) {
+          theVaultMaxAmount = issuableTokens.toBn();
+          break;
+        }
```

Comment:
@savudani8 @nud3l 
I guess it could be:
```suggestion
  const issuableTokens = theVaults.values()[0];
  theVaultMaxAmount = issuableTokens.toBn();
```

Reply:
Yep, that makes sense!